script.js

```diff
@@ -0,0 +1,204 @@
+    {
+        Result();
+    }
+    //second row
+    else if ((grid[3].classList.contains('x') && grid[4].classList.contains('x') && grid[5].classList.contains('x')) || (grid[3].classList.contains('o') && grid[4].classList.contains('o') && grid[5].classList.contains('o')))
+    {
+        Result();
+    }
+    //third row
+    else if ((grid[6].classList.contains('x') && grid[7].classList.contains('x') && grid[8].classList.contains('x')) || (grid[6].classList.contains('o') && grid[7].classList.contains('o') && grid[8].classList.contains('o')))
+    {
+        Result();
+    }
+    //first diagonal
+    else if ((grid[0].classList.contains('x') && grid[4].classList.contains('x') && grid[8].classList.contains('x')) || (grid[0].classList.contains('o') && grid[4].classList.contains('o') && grid[8].classList.contains('o')))
+    {
+        Result();
+    }
+    //second diagonal
+    else if ((grid[2].classList.contains('x') && grid[4].classList.contains('x') && grid[6].classList.contains('x')) || (grid[2].classList.contains('o') && grid[4].classList.contains('o') && grid[6].classList.contains('o')))
+    {
+        Result();
+    }
+    //draw
+    else
+    {
+        //loop through grid and check if its filled
+        for (let i = 0; i < grid.length; i++)
+        {
+            //break out of loop if a square is not marked
+            if (!grid[i].classList.contains('marked'))
+            {
+                break;
+            }
+            //grid filled
+            if (i === 8)
+            {
+                gridFilled = true;
+            }
+        }
+
+        //grid is filled and there is no win
+        if (gridFilled)
+        {
+            result.textContent = 'It\'s a draw!';
+            gameOver = true;
+        }
+    }
+}
+
+
+/***************** MAIN *****************/
+
+//start game
+Initialize();
+
+//reset button
+reset.addEventListener('click', Reset);
+
+//add event listeners to all squares in grid
+for (let square of grid)
+{
+    square.addEventListener('click', () =>
+    {
+        //check if game is over
+        if (!gameOver)
+        {
+            //check if box is already marked
+            if (!square.classList.contains('marked'))
+            {
+                //add marked class when square is clicked
+                square.classList.add('marked');
+
+                //mark with x or o based on current turn, check board for draw/win and pass turn
+                if (currentTurn == 'x')
+                {
+                    square.classList.add('x');
+                    square.textContent = 'X';
+                    CheckGrid();
+                    PassTurn();
+                }
+                else
+                {
+                    square.classList.add('o');
+                    square.textContent = 'O';
+                    CheckGrid();
+                    PassTurn();
+                }
+            }
+        }
+    })
+}
```

Comment:
Normally nested if statements can get convoluted but since you've labeled your variables and logic extremely well this still makes it very easy to follow so great job on that.